### PR TITLE
Add comprehensive describe/it/should test framework

### PR DESCRIPTION
**Pull request by paul-hammant**
Implemented a complete BDD-style testing framework for ChrysaLisp with extensive features, examples, and documentation.

Core Framework (lib/test/test.inc):
- BDD-style test organization with describe/it blocks
- Skip functionality (xdescribe/xit)
- 17 comprehensive assertions (equality, boolean, nil, comparison, collections, error handling)
- Colored terminal output with symbols (✓/✗/○)
- Detailed failure reporting
- Configurable filtering and verbosity
- Test state management using Fmap

Test Runner (cmd/test-runner.lisp):
- Automatic test file discovery
- Pattern-based filtering (-f flag)
- Verbose mode (-v flag)
- Color control (-n flag)
- Custom file patterns (-p flag)
- List tests mode (-l flag)
- Exit codes for CI/CD integration (0=pass, 1=fail)

Examples (examples/test/):
- basic_test.lisp: Fundamental patterns and assertions
- collections_test.lisp: Sequences and collection operations
- advanced_test.lisp: Error handling, recursion, edge cases
- four_horsemen_test.lisp: Deep dive into map/filter/reduce/some
- smoke_test.lisp: Quick sanity check
- ~80-90 total tests demonstrating all features

Documentation:
- docs/test_framework.md: Comprehensive guide with API reference
- docs/test_framework_quick_reference.md: Quick reference cheat sheet
- examples/test/README.md: Example file guide and learning path
- TEST_FRAMEWORK_SUMMARY.md: Implementation summary

Features:
- Philosophy-aligned with ChrysaLisp (iteration, Four Horsemen, O(1))
- BDD best practices (readable, organized, focused)
- Excellent developer experience (fast, clear, flexible)
- Production-ready with ~1,810 lines of code and documentation

Usage:
  ./run_tui.sh -n 1 cmd/test-runner.lisp [options] [files...]

### CODE REVIEW

File: cmd/test-runner.lisp [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+;;;;;;;;;;;;;;;;;;;;;;;;
+; Options
+;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defq opt-filter :nil)
+(defq opt-verbose :nil)
+(defq opt-no-color :nil)
+(defq opt-pattern "examples/test/**/*_test.lisp")
+(defq opt-list :nil)
```
Comment (vygr) [inline cmd/test-runner.lisp]: Variable names style guide problems.
